pkg/cluster/cluster.go

```diff
@@ -315,36 +314,36 @@ func New(ctx context.Context, log *logrus.Entry, _env env.Interface, db database
 			return nil, err
 		}
 
-		msiDataplaneClientOptions, err := _env.MsiDataplaneClientOptions(msiResourceId)
-		if err != nil {
-			return nil, err
-		}
+		var msiDataplane dataplane.ClientFactory
+		if _env.FeatureIsSet(env.FeatureUseMockMsiRp) {
+			msiDataplane = _env.MockMSIResponses(msiResourceId)
```

Comment:
We need to mock out the MSI dataplane, so we can just provide a different implementation for the interface. The previous stub approach of injecting data through an HTTP transport was fragile - it, for one, did not make any distinction between which method was being called!